### ISSUE TITLE
no hosts rule is made at install or start

### ISSUE THREAD

**Issue by olbrew** [nlf/dlite#87]
My issue is related to #57 but occurs for me after a fresh install.
I installed `dlite` via `brew`. I did `sudo dlite install` and `dlite start` but there is never any rule added to my `/etc/hosts` file. I tried rebooting, reinstalling, ... to no avail.
I'm on OS X 10.11.3 should that matter.
Console says not much more than `Service exited with abnormal code: 1`.
My `dhcpd_leases` also doesn't have an entry with dlite's `config.json`.

**Comment by nlf** [comment]
what happens if you run `sudo dlite daemon`?

**Comment by olbrew** [comment]
```
COM1 connected to /dev/ttys001
rdmsr to register 0x570 on vcpu 0
rdmsr to register 0x34 on vcpu 0
Failed to find an IP for the virtual machine in 15 seconds
```

**Comment by 14to9** [comment]
Getting identical behavior on coworker's machine.  OSX 10.10.5

dlite [1.1.3] fails to start entirely after install.

**Comment by 14to9** [comment]
This looks to us like it's related to OSX Firewall settings.  Disable firewall, run install again results for us in both `dhcpd_leases` and `/etc/hosts` being updated as expected.  dlite then starts normally with these assignments in place.

Theory:  When the `dhcpd_leases` file isn't written to, perhaps it's because the request for IP address is getting blocked.  

@nlf are you running particular Firewall settings?

**Comment by ylafrance** [comment]
I'm 14to9 coworker. I want to add that the firewall can be put back on after "dlite start" and it's still working.

Update: Not true after all. docker-compose up couldn't complete with the firewall on. Sorry about that.

**Comment by olbrew** [comment]
@ylafrance did you completely disable the OS X firewall? Mine is on, but not in `block all incoming connections` mode. Normally it should popup a dialog with network request.

Anyway I tried disabling and reinstalling but it still doesn't create any `hosts` or `dhcpd_leases` entries unfortunately.

**Comment by ylafrance** [comment]
@olbrew My firewall is indeed completely disabled
- I uninstalled **dlite** `dlite uninstall`
- disabled my firewall
- restarted my computer
- installed **dlite** `dlite install`
- started **dlite** `dlite start`
- verified it was running using `ps -aef | grep dlite`

I think that's all

The `/etc/exports` file had the following lines added:

```
# BEGIN: dlite
/Users -network 192.168.64.0 -mask 255.255.255.0 -alldirs -maproot=root:wheel
# END: dlite
```

The `/etc/hosts` file had the following line added:
`192.168.64.5 local.docker # added by dlite`
And the `/private/var/db/dhcpd_leases` file now had lines for `192.168.64.5`

Note that after successful installation, I had NFS issues until the exports line was changed to:

```
# BEGIN: dlite
/Users -network 192.168.64.0 -mask 255.255.255.0 -alldirs -mapall=root:wheel
# END: dlite
```

**Comment by olbrew** [comment]
Awesome, that did indeed do it. Thanks!
Seems rebooting between after the firewall was a necessary step.

**Comment by nlf** [comment]
so am i right in thinking that if i document that the firewall should be disabled and the machine rebooted before installation everything would work ok?

**Comment by ruimarinho** [comment]
What about checking if the firewall is enabled?

``` sh
test $(defaults read /Library/Preferences/com.apple.alf globalstate) -ge 1
```

And perhaps ask the user if it can be disabled during installation?

``` sh
defaults write /Library/Preferences/com.apple.alf globalstate -int 0
```

**Comment by olbrew** [comment]
Problem is that it needs to stay disabled for Docker itself to work properly :(

**Comment by nlf** [comment]
i'll have to tinker and see if i can figure out a set of rules that will allow things to work with the firewall enabled

**Comment by supersophie** [comment]
Noticed no entry in /etc/hosts nor any socket created in /var/run/ 
Entry added and socket created after I ran below

$ sudo dlite daemon
COM1 connected to /dev/ttys001
XHYVE: vlapic callout at 0x702b.0xacc252dc657d50d0, expected at 0x702b.#ad1c542067438ceb
rdmsr to register 0x570 on vcpu 0
rdmsr to register 0x34 on vcpu 0

OSx Version 10.11.03

Logging in and out did not start the virtual machine.

Had to run 'dlite deamon' for this to start.
I could drop think into .bash_profile, but don't think this would be correct :) What do you recommend?

**Comment by nlf** [comment]
the launchd agent should be working.. what happens if you run 

``` bash
launchctl unload ~/Library/LaunchAgents/local.dlite.plist
launchctl load ~/Library/LaunchAgents/local.dlite.plist
```

do you get dlite running after that?

**Comment by nlf** [comment]
we'll need to test to verify, but i think the switch to 9p over NFS will resolve the original issue here.

if any of you are keeping your firewall enabled and want to help make sure this gets resolved, check out issue #135 for details
